docs/book/src/images/kubeadm-control-plane-machines-resources.plantuml

```diff
@@ -0,0 +1,50 @@
+@startuml kubeadm-control-plane-machines-resources
+
+skinparam nodesep 80
+skinparam ranksep 80
+skinparam object {
+    backgroundColor<< Infrastructure provider CRD >> #D9EAD3
+    backgroundColor<< ControlPlane CRD >> #EBF6F7
+    backgroundColor<< Bootstrap provider CRD >> #FEF4F4
+}
+
+' Cluster API CRD
+object Cluster << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureRef</font>
+    spec.controlPlaneRef
+}
+object MachineHealthCheck << Cluster API CRD >> {
+    spec.clusterName
+}
+object Machine << Cluster API CRD >> {
+    <font color="Green">spec.infrastructureRef</font>
+    spec.bootstrap.configRef
+}
+
+' ControlPlane CRD
+object KubeadmControlPlane << ControlPlane CRD >> {
+    <font color="Green">spec.infrastructureTemplate</font>
+    <font color="Red">spec.BarBoostrapConfigSpec</font>
+}
+
+' Bootstrap provider CRD
+object BarBootstrapConfig << Bootstrap provider CRD >> 
+
+' Infrastructure provider CRD
+object FooCluster << Infrastructure provider CRD >>
+object FooMachineTemplate << Infrastructure provider CRD >>
```

Comment:
It would be great if we can keep FooMachineTemplate and FooMachine on the same side of the graph..